Storage/src/StreamWrapper.php

```diff
@@ -127,6 +127,29 @@ public function stream_set_option()
         return false;
     }
 
+    /**
+     * This is called when touch is used on a stream. See:
+     * https://www.php.net/manual/en/streamwrapper.stream-metadata.php
+     */
+    public function stream_metadata($path, $option, $value)
+    {
+        if ($option == STREAM_META_TOUCH) {
+            $this->openPath($path);
+            $object = $this->bucket->object($this->file);
+            try {
+                if (!$object->exists()) {
```

Comment:
Can we move this logic to a function named `touch`? Eg: [mkdir](https://github.com/googleapis/google-cloud-php/blob/8fec964fbb2b41571b27a7b5843bcf1477a7bf60/Storage/src/StreamWrapper.php#L474)

Reply:
[82f759c](https://github.com/googleapis/google-cloud-php/pull/7144/commits/82f759cbe83275846757eca017f594affcbbfa64)